src/jit/compiler.hpp

```diff
@@ -757,13 +757,26 @@ inline
   signed __int16    getI2LittleEndian(const BYTE * ptr)
 { return * (UNALIGNED signed __int16 *)ptr; }
 
+#if defined(_ARM_) && defined(__linux__) && !defined (DEBUG)
```

Comment:
> Could you please use the GET_UNALIGNED_VAL32 and GET_UNALIGNED_VAL64 in 
> these functions instead of this change as @jkotas has suggested before? It would make 
> this change cleaner without any need for ifdefs.

Yes. I have modified the exiting PR. I will  again send  the PR as soon as I verify the effect of  new PR with the CoreCLR's unit-test.